staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -353,6 +386,19 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 		token: cred.Status.Token,
 	}
 	if cred.Status.ClientKeyData != "" && cred.Status.ClientCertificateData != "" {
+		// Check cert expiration time.
+		certBlock, _ := pem.Decode([]byte(cred.Status.ClientCertificateData))
+		if certBlock == nil {
+			return errors.New("can't parse client certificate returned by exec plugin as PEM block")
+		}
+		x509Cert, err := x509.ParseCertificate(certBlock.Bytes)
```

Comment:
Use the cert on line 402 so you don't have to parse the pem twice.

Reply:
You mean manually parse `ClientKeyData` and build `tls.Certificate` via a literal?
It feels like more work and easy to miss some validations that `tls.X509KeyPair` does